### ISSUE TITLE
Routes in actions not in traditional app-structure aren't loaded

### ISSUE THREAD

**Issue by kendugu** [lorisleiva/laravel-actions#133]
When loading routes, `ActionManager` uses Laravel's `Util::loadClasses()` which in turn uses this method:

```
public static function getClassnameFromRealpath(string $realpath): string
    {
        return Container::getInstance()->getNamespace() . str_replace(
            ['/', '.php'],
            ['\\', ''],
            Str::after($realpath, realpath(app_path()).DIRECTORY_SEPARATOR)
        );
    }
```
...to turn filenames into classnames. There is an issue for this on the Laravel side in that Laravel assumes (through the use of `app_path()`) that the class exists in the default folder structure (i.e. `App/`). 

The result of this is that routes inside actions in a modular or package-based architecture won't be loaded because Laravel assumes a "normal" folder structure. For example, I have the following package action:

`/project/packages/package-name/src/Actions/DoSomethingAction.php`

which `Util::loadClasses` translates to a class path of:

`App\project\packages\package-name\src\Actions\DoSomethingAction`

which will not be found or loaded.

This is more an issue of the Laravel framework than Laravel Actions, but in case someone else has the issue I thought I'd post about it here.

**Comment by lorisleiva** [comment]
Hi there 👋 

Yeah I noticed this myself when trying different folder structures — e.g. `Domain\...`.

The trouble is, even on the framework, we need to know how to convert a base file path into a root namespace. We could probably use the `composer.json` to get that information but it feel like an overkill when most use cases will be using the `App` namespace.

That being said I have recently extracted that logic into its own package: https://github.com/lorisleiva/lody

The only reason why I haven't used it on Laravel Actions yet is that I've made the package PHP8 only whereas at the moment Laravel Actions also supports PHP7.4. As soon as Laravel Actions drops support for the latter, I'll make the swap.

All that to say, these optimisations could then be implemented on the Lody package since it's a clearly defined problem and that would allow it to also work on different frameworks/folder structures.

In any case, I'll keep this open until we sort something better because it's still very much on my mind.

**Comment by WorthLMS** [comment]
How would you use Lody to resolve this issue?

my tests work fine when using postJson($endpoint, $data)... but my browser and Postman fail to see the route....

I don't see what postJson() tests do differently... but whatever... I need this to work... how do I get it to recognize my routes?

**Comment by lorisleiva** [comment]
This has been implemented in the [Lody package](https://github.com/lorisleiva/lody) which is now being used by Laravel Actions to register routes and commands. This means Laravel Actions now supports registering routes and commands in any namespace — not only in the `App` namespace — and even in any namespace within your `vendor` directory. It uses your `vendor/composer/autoload_psr4.php` file which uses the PSR-4 configurations present in your `composer.json`.

This was released in [`v2.2.0`](https://github.com/lorisleiva/laravel-actions/releases/tag/v2.2.0). 🚀
